### PR TITLE
Make some gp likelihoods more stable

### PR DESCRIPTION
**Pull request by fehiepsi**
Addresses #1996. When `response_function` takes its default value, what it does is to convert `logits` to `probs` in Binary/Categorical distributions. This PR skips applying `response_function` for the default case to take advantage of the numerical stability when using these distributions with `logits` arg.

### CODE REVIEW

File: examples/contrib/gp/sv-dkl.py [hunk 1]
```diff
@@ -143,7 +143,7 @@ def main(args):
     # Turns on "whiten" flag will help optimization for variational models.
     gpmodule = gp.models.VariationalSparseGP(X=Xu, y=None, kernel=deep_kernel, Xu=Xu,
                                              likelihood=likelihood, latent_shape=latent_shape,
-                                             num_data=60000, whiten=True)
+                                             num_data=60000, whiten=True, jitter=2e-6)
```
Comment (fehiepsi) [inline examples/contrib/gp/sv-dkl.py]: When I run this example again, I get Cholesky error so I increase jitter a little bit. The reason might come from the use of a different system or the changes in PyTorch linear algebra module.

File: pyro/contrib/gp/likelihoods/binary.py [hunk 2]
```diff
@@ -40,9 +40,11 @@ def forward(self, f_loc, f_var, y=None):
...
-
-        y_dist = dist.Bernoulli(f_res)
+        if self.response_function is torch.sigmoid:
+            y_dist = dist.Bernoulli(logits=f)
+        else:
+            f_res = self.response_function(f)
+            y_dist = dist.Bernoulli(f_res)
         if y is not None:
-            y_dist = y_dist.expand_by(y.shape[:-f_res.dim()]).to_event(y.dim())
+            y_dist = y_dist.expand_by(y.shape[:-f.dim()]).to_event(y.dim())
```
Comment (fritzo) [inline pyro/contrib/gp/likelihoods/binary.py]: Can you confirm that `f.dim()` is never 0? Because if `f.dim()` is ever zero, you'll need the more complex expression `y.shape[:y.dim() - f.dim()]`.

Comment (fehiepsi) [reply pyro/contrib/gp/likelihoods/binary.py]: Yes. Kernels only allow [inputs to be 1D or 2D](https://github.com/pyro-ppl/pyro/blob/dev/pyro/contrib/gp/kernels/kernel.py#L52), so `f` will be at least 1D, even when we are applying GP on 1 data point.
